libvast/src/system/tracker.cpp

```diff
@@ -166,6 +164,13 @@ void register_component(scheduled_actor* self, tracker_state& st,
   }
 }
 
+// Checks whether a component can be spawned at most once.
+bool is_singleton(const std::string& component) {
+  static detail::flat_set<std::string> singletons = {"archive", "index",
+                                                     "consensus"};
+  return singletons.find(component) != singletons.end();
```

Comment:
I would avoid bringing in such a big dependency here, how about:
``` c++
  const char* singletons[] = {"archive", "index", "consensus"};
  return std::any_of(std::begin(singletons), std::end(singletons),
                     [&](const char* lhs) { return lhs == component; });
```

Reply:
Fair 'nuf. I've incorporated your suggestion.